Sources/Testing/Events/Recorder/Event.AdvancedConsoleOutputRecorder.swift

```diff
@@ -64,71 +142,716 @@ extension Event {
+          
+          output += "\(issuePrefix)\(issueTreePrefix)\(issueIcon) \(issueDescription)\n"
+          
+          // Add source location
+          if let sourceLocation = issue.sourceLocation {
+            let locationPrefix = issuePrefix + (isLastIssue ? "   " : "\(_treeVertical)  ")
+            output += "\(locationPrefix)At \(sourceLocation.fileName):\(sourceLocation.line):\(sourceLocation.column)\n"
+          }
+        }
+      }
+    }
+    
+    return output
+  }
+  
+  /// Get the status icon for a test result.
+  ///
+  /// - Parameters:
+  ///   - result: The test result.
+  /// - Returns: The appropriate symbol string.
+  private func _getStatusIcon(for result: _TestResult) -> String {
+    switch result {
+    case .passed:
+      return Event.Symbol.pass(knownIssueCount: 0).stringValue(options: options.base)
+    case .failed:
+      return Event.Symbol.fail.stringValue(options: options.base)
+    case .skipped:
+      return Event.Symbol.skip.stringValue(options: options.base)
+    }
+  }
+  
+  /// Format a duration in seconds with exactly 2 decimal places.
+  ///
+  /// - Parameter duration: The duration to format.
+  /// - Returns: A formatted duration string (e.g., "1.80s", "0.05s").
+  private func _formatDuration(_ duration: Double) -> String {
+    // Always format to exactly 2 decimal places
+    let wholePart = Int(duration)
+    let fractionalPart = Int((duration - Double(wholePart)) * 100 + 0.5) // Round to nearest hundredth
+    
+    // Handle rounding overflow (e.g., 0.999 -> 1.00)
+    if fractionalPart >= 100 {
+      return "\(wholePart + 1).00s"
+    } else {
+      let fractionalString = fractionalPart < 10 ? "0\(fractionalPart)" : "\(fractionalPart)"
+      return "\(wholePart).\(fractionalString)s"
+    }
+  }
+  
+  /// Pad a test line with right-aligned duration.
+  ///
+  /// - Parameters:
+  ///   - testLine: The test line to pad.
+  ///   - duration: The duration string.
+  /// - Returns: The padded test line with right-aligned duration.
+  private func _padWithDuration(_ testLine: String, duration: String) -> String {
+    if duration.isEmpty {
+      return testLine
+    }
+    
+    let targetWidth = 100
+    let rightPart = "(\(duration))"
+    let totalLeftLength = testLine.count
+    let totalRightLength = rightPart.count
+    
+    if totalLeftLength + totalRightLength + 5 < targetWidth {
+      let paddingLength = targetWidth - totalLeftLength - totalRightLength
+      return "\(testLine)\(String(repeating: " ", count: paddingLength))\(rightPart)"
+    } else {
+      return "\(testLine) \(rightPart)"
+    }
+  }
+  
+  /// Check if a string contains Swift file pattern without using Foundation.
+  ///
+  /// - Parameters:
+  ///   - string: The string to check.
+  /// - Returns: True if the string contains ".swift:" pattern.
+  private func _containsSwiftFile(_ string: String) -> Bool {
+    let target = ".swift:"
```

Comment:
This string can be present in any component of a test ID.

The last component of a test ID string should always be the filename, so we don't really need to check each component. If you're doing this because of some limitation in the JSON data, can you clarify so we can improve it? Thanks!